### PR TITLE
object-manipulation-2 | Richard Watanabe

### PR DESCRIPTION
**Pull request by Richard-Watanabe**
Screen Shot 2021-07-19 at 9 35 17 PM

### CODE REVIEW

File: object-manipulation-2/src/omit.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// Return the storage.
+
+function omit(source, keys) {
+  var newObject = {};
+  for (var property in source) {
+    for (var i = 0; i < keys.length; i++) {
+      var value = source[property];
+      newObject[property] = value;
+      delete newObject[keys[i]];
+    }
```
Comment (codingcodymiller) [inline object-manipulation-2/src/omit.js:15]: I believe you can do this loop to delete properties after the first loop to create the properties runs. As of right now, you are running this full loop for every property in the object as you create it, and that is a lot of extra operations for the computer.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by Richard-Watanabe** [comment]
Screen Shot 2021-07-20 at 6 00 07 PM

File: object-manipulation-2/src/omit.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// Return the storage.
+
+function omit(source, keys) {
+  var newObject = {};
+  for (var property in source) {
+    var value = source[property];
+    newObject[property] = value;
+    for (var i = 0; i < keys.length; i++) {
+      delete newObject[keys[i]];
+    }
```
Comment (codingcodymiller) [inline object-manipulation-2/src/omit.js:15]: This is still inside the `for in` loop, try moving it below the `for in` loop and see if it still works 🤞

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by Richard-Watanabe** [comment]
Screen Shot 2021-07-21 at 9 22 13 PM

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**